core/src/main/java/org/apache/carbondata/core/datastorage/store/compression/type/UnCompressNonDecimalMaxMinInt.java

```diff
@@ -93,7 +94,9 @@
       if (value[i] == 0) {
         vals[i] = maxValue;
       } else {
-        vals[i] = (maxValue - value[i]) / Math.pow(10, decimal);
+        BigDecimal diff = new BigDecimal(Double.toString(value[i] / Math.pow(10, decimal)));
```

Comment:
use BigDecimal.valueOf(double) instead of converting to string for performance and dataloss concerns.refer http://www.javaworld.com/article/2073176/caution--double-to-bigdecimal-in-java.html